src/layout/components/Header/index.tsx

```diff
@@ -3,12 +3,25 @@ import { NavLink } from 'react-router-dom';
 
 import NavDropdown from '../../containers/NavDropdown';
 import NavProfile from '../../containers/NavProfile';
-
+import Icon from '@ant-design/icons';
 import logo from '../../../assets/graphics/logo.png';
-
+import './custom.less';
+import { Switch } from 'antd';
 import './style.less';
+import ThemeContext from '../../../styles/ThemeContext/themeContext';
+import { ReactComponent as MoonIcon } from '../../../assets/icons/moon.svg';
+import { ReactComponent as SunIcon } from '../../../assets/icons/sun.svg';
+
 
 const Header: React.FC = () => {
+
+  // gets theme and toggleTheme function from the ThemeContext
+  const {theme, toggleTheme } = React.useContext(ThemeContext);
+
+  const HandleModeSwitch = () => {
+    toggleTheme?.();
```

Comment:
`toggleTheme` should always be defined. may need to add types to `ThemeContext` or something if typescript is throwing errors.